plutus-core/executables/uplc/Main.hs

```diff
@@ -204,6 +204,88 @@ runEval (EvalOptions inp ifmt printMode budgetMode traceMode outputMode timingMo
                         None -> pure ()
                         _    -> writeToFileOrStd outputMode (T.unpack (T.intercalate "\n" logs))
 
+
+-- Experimental optimisation involving force/delay.
```

Comment:
I just stuck all this stuff in here because I'm not intending to merge it and this was the easiest place to put it.